pkg/image/image.go

```diff
@@ -46,15 +53,17 @@ type UDF struct {
 }
 
 // Create makes the required call to UDF to
-func (u *UDF) Create(release, channel, arch string) (path string, err error) {
-	tmpFileName, _ := u.cli.ExecCommand("mktemp")
+func (u *UDF) Create(release, channel, arch string, ver int) (path string, err error) {
+	tmpDirName, _ := u.cli.ExecCommand("mktemp -d")
+	tmpFileName := filepath.Join(tmpDirName, outputFileName)
 
 	var archFlag string
 	if arch == "arm" {
 		archFlag = "--oem beagleblack"
 	}
 	_, err = u.cli.ExecCommand("sudo", "ubuntu-device-flash", "core", release,
-		"--channel", channel, "--developer-mode", archFlag, "-o", tmpFileName)
+		"--revision", strconv.Itoa(ver), "--channel", channel, "--developer-mode",
```

Comment:
I'm not sure if we need to use negative numbers in here, but if we do, remember that we need to pass something like --revision=-1 instead of --revision -1.